engine/resources/exec.go

```diff
@@ -800,3 +821,21 @@ func (obj *wrapWriter) Write(p []byte) (int, error) {
 func (obj *wrapWriter) String() string {
 	return obj.Buffer.String()
 }
+
+// checks that environment variable name is valid
+func isNameValid(varName string) error {
+	if varName == "" {
+		return fmt.Errorf("variable name cannot be an empty string")
+	}
+	i := 0
+	for i = 0; i < len(varName); i++ {
+		c := varName[i]
+		if i == 0 && '0' <= c && c <= '9' {
```

Comment:
This is yucky. Just use strings.HasPrefix outside the loop.

Reply:
Maybe I'm missing something, but wouldn't I have to loop through characters 0-9 if I used strings.HasPrefix? That doesn't sound like the simplest solution.

Reply:
come to think of it I could just use regex

Reply:
Am I wrong that it would be more simple to not add a second loop?

Reply:
There are more readable ways to write this, but after looking at it further, they're less efficient. You can keep it as is, but at least use the loop `range` construct.